File: Lecture4/Ex44_RandomNumberGen.java
public class Ex44_RandomNumberGen {
    public static void main(String[] args) {

        boolean flag_minus18 = false;
        boolean flag_9 = false;

        int randomNumber, cnt = 1;
        int first_minus18 = 0, first_9 = 0;

        while ( !(flag_minus18 && flag_9) ) {
            randomNumber = (int) (Math.random() * 28 ) - 18;

            if ( randomNumber == -18 ) {
                if ( !flag_minus18 ) first_minus18 = cnt;
                System.out.println( ">> Found -18 at " + cnt);
                flag_minus18 = true;
            } else if( randomNumber == 9 ) {
                if ( !flag_9 ) first_9 = cnt;
                System.out.println( ">> Found 9 at " + cnt);
                flag_9 = true;
            }

            cnt++;
        }

        System.out.println( "The first appearance of -18 was at " + first_minus18 );
        System.out.println( "The first appearance of 9 was at " + first_9 );

    }
}

File: Lecture4/Ex44_RandomNumberGen.c
#include <stdio.h>
#include <stdlib.h>
#include <time.h>

int main(void) {

    srand(time(NULL));

    int flag_minus18 = 0;
    int flag_9 = 0;

    int randomNumber, cnt = 1;
    int first_minus18 = 0, first_9 = 0;

    while ( !(flag_minus18 && flag_9) ) {
        randomNumber = rand() % 28 - 18;

        if ( randomNumber == -18 ) {
            if ( !flag_minus18 ) first_minus18 = cnt;
            printf( ">> Found -18 at %d\n", cnt);
            flag_minus18 = 1;
        } else if( randomNumber == 9 ) {
            if ( !flag_9 ) first_9 = cnt;
            printf( ">> Found 9 at %d\n", cnt);
            flag_9 = 1;
        }

        cnt++;
    }

    printf( "The first appearance of -18 was at %d\n", first_minus18 );
    printf( "The first appearance of 9 was at %d\n", first_9 );

    return 0;
}
